File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeNthFromEnd(struct ListNode* head, int n){
    struct ListNode *temp=head;
    if (head==NULL){
        return NULL;
    }
    int length=0;
    while(temp!=NULL){
        temp=temp->next;
        length+=1;
    }
    if (length==1){
        return NULL;
    }
    int rem_ind=length-n-1;
    int count=0;
    struct ListNode *temp1=head;
    if(rem_ind==-1){
        return head->next;
    }
    while(count<rem_ind){
        temp1=temp1->next;
        count+=1;
    }
    temp1->next=temp1->next->next;
    return head;

}

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        leng=0
        temp=head
        while temp:
            temp=temp.next
            leng+=1
        a=leng-n-1
        if leng==n:
            if head.next:
                return head.next
            return None
        temp=head
        while a:
            temp=temp.next
            a-=1
        temp.next=temp.next.next
        return head